zmq.hpp

```diff
@@ -2224,6 +2224,53 @@ class monitor_t
         on_monitor_started();
     }
 
+#if ZMQ_VERSION_MAJOR >= 4
+    bool get_event(zmq_event_t& eventMsg, std::string& address, zmq::recv_flags flags = zmq::recv_flags::none)
+    {
+        assert(_monitor_socket);
+
+        eventMsg.event = 0;
+        eventMsg.value = 0;
+        address = std::string();
+
+        {
+            message_t msg;
+            int rc = zmq_msg_recv(msg.handle(), _monitor_socket.handle(),
+                static_cast<int>(flags));
+
+            if (rc == -1)
+            {
+                if (zmq_errno() == ETERM || zmq_errno() == EAGAIN)
+                    return false;
+                else
+                    throw error_t();
+            }
+
+            const char *data = msg.data<char>();
+            memcpy(&eventMsg.event, data, sizeof(uint16_t));
+            data += sizeof(uint16_t);
+            memcpy(&eventMsg.value, data, sizeof(int32_t));
+        }
+
+        message_t addrMsg;
+        int rc = zmq_msg_recv(addrMsg.handle(), _monitor_socket.handle(),
```

Comment:
Like above, can't this use `_monitor_socket.recv`?

Reply:
See prev comment

> Thanks for the extensions, I think these will be really useful.
> 
> Please see my comments on individual issues.
> 
> Also, please remember to reformat all commits with `clang-format`.
> 

Could you add this to the contribution guidelines of the project?

> Most of the travis builds are failing right now, could you give them a look and fix them, please?

I don't like to put effort on something that will be not merged. I was waiting for a green thumb before putting more work on this, as I did not asked anyone about the design of this PR ;)

Thanks for reviewing, will try to solve all of those next week. BR